src/app/chat/view.nim

```diff
@@ -152,7 +155,7 @@ QtObject:
       msg.alias = self.status.chat.getUserName(msg.fromAuthor, msg.alias)
       self.messageList[msg.chatId].add(msg)
       self.messagePushed()
-      if msg.chatId != self.activeChannel.id:
+      if msg.chatId != self.activeChannel.id and self.channelOpenTime.hasKey(msg.chatId) and self.channelOpenTime[msg.chatId] < msg.timestamp.parseFloat.fromUnixFloat.toUnix:
```

Comment:
Have you considered using `self.channelOpenTime.getOrDefault`, ie:
```suggestion
      if msg.chatId != self.activeChannel.id and self.channelOpenTime.getOrDefault(msg.chatId, high(int64)) < msg.timestamp.parseFloat.fromUnixFloat.toUnix:
```

Reply:
nice!